pythonforandroid/recipes/python3/__init__.py

```diff
@@ -323,12 +340,17 @@ def build_arch(self, arch):
                     *(' '.join(self.configure_args).format(
                                     android_host=env['HOSTARCH'],
                                     android_build=android_build,
+                                    python_host_bin=join(self.get_recipe(
+                                        'host' + self.name, self.ctx
+                                    ).get_path_to_python(), "python3"),
                                     prefix=sys_prefix,
                                     exec_prefix=sys_exec_prefix)).split(' '),
                     _env=env)
 
+            # disable parallel build on 3.11 to fix:
+            # ld: error: unable to find library -lpython3.11
             shprint(
-                sh.make, 'all', '-j', str(cpu_count()),
+                sh.make, 'all', '-j', (str(cpu_count()) if not version_starts_with("3.11") else "1"),
```

Comment:
This issue on CPython upstream look similar: https://github.com/python/cpython/issues/101295 (Not the same, but I suspect the root cause is the same)

IMHO, let's just avoid this issue by keeping it simple, in all cases, since Python3.11 will be the default:

```python
shprint(sh.make, 'all')
```
Also, can you please add a reference to the above-mentioned issue?

Something like:
```python
# Python build does not seem to play well with make -j option from Python 3.11 and onwards
# Before losing some time, please check issue
# https://github.com/python/cpython/issues/101295 , as the root cause looks similar
```